### DISCUSSION TITLE
Retrieving tar.gz artifact from source-controller/gitrepository resource

### DISCUSSION THREAD

**Discussion by dariuslwh95** [fluxcd/flux2#3988] [Q&A]
hi, have seen in https://fluxcd.io/flux/components/source/gitrepositories/
"The Artifact file is a gzip compressed TAR archive (<commit sha>.tar.gz), and can be retrieved in-cluster from the .status.artifact.url HTTP address."

Have been having difficulty doing so from e.g. a node debug pod in the flux-system namespace / the kustomization-controller pod using curl / wget commands. 

/ $ wget http://source-controller.flux-system.svc.cluster.local./gitrepository/flux-system/cnocp-np/eaf99b4f35a68fc8e05d56837ff215f890eca944.tar.gz
Connecting to source-controller.flux-system.svc.cluster.local. (xxxx)
wget: server returned error: HTTP/1.1 404 Not Found

/ $ ping http://source-controller.flux-system.svc.cluster.local./gitrepository/flux-system/csre-bau-np/latest.tar.gz
ping: bad port spec 'http://source-controller.flux-system.svc.cluster.local./gitrepository/flux-system/csre-bau-np/latest.tar.gz'

my environment is openshift 4x, do you have any tips for retrieving this tarball? there are no issues in reconcilliation within the cluster.

**Comment by makkes** [comment]
Since the source-controller apparently returns a 404 there's no connectivity issue. You'll get the artifact URL from the `GitRepository`'s `.status.artifact.url` field, e.g. with ` kubectl get gitrepo -n flux-system cnocp-np -o jsonpath={.status.artifact.url}`.
